cache/FIFO_FH.h

```diff
@@ -606,12 +620,13 @@ bool FIFO_FHCache<TKey, TValue, THash>::find(TValue& ac,
     // we can re-check this and remove it if possible
 
     // Generally when we found the thing in frozen cache
-    if(m_fasthash->find(key, ac) && (ac != nullptr))
+    std::unique_lock<ListMutex> lockMap(m_mapMutex);
+    if(m_fasthash->find(key, ac) && (ac != nullptr) && (chunkMapper[key] > globalCounter))
```

Comment:
This may be dependent on the data structure we end up using for chunkMapper, but is the ordering to check m_fashhash before chunkMapper intentional?